StackExchange.Profiling/Timing.cs

```diff
@@ -299,6 +325,7 @@ internal void RemoveCustomTiming(string category, CustomTiming customTiming)
         }
 
         private readonly object _lockObject = new object();
```

Comment:
I think the existing method acquires _lockObject first and then the specific list if it is not null, i dont think this makes sense, since there are many cases where racing might occur.